code/game/objects/structures/urban.dm

```diff
@@ -351,16 +351,21 @@
 	can_be_unanchored = FALSE
 	layer = ABOVE_ALL_MOB_LAYER
 
-/obj/structure/chainlink_fence/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
-	if(istype(mover,/obj/item/projectile))
-		return TRUE
-	if(!istype(mover) || mover.checkpass(PASSRAILING))
+/obj/structure/chainlink_fence/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)//So bullets will fly over and stuff.
+	if(air_group || (height==0))
 		return TRUE
-	if(mover.throwing)
+	if(istype(mover, /obj/item/projectile))
+		var/obj/item/projectile/P = mover
+		if(P.original == src)
+			return FALSE
+		if(P.firer && Adjacent(P.firer))
+			return TRUE
+		return prob(35)
+	if(isliving(mover))
+		return FALSE
+	if(istype(mover) && mover.checkpass(PASSTABLE))
```

Comment:
Passtable is probably a bit goofy on a tall chainlink fence

Reply:
This was taken from the other chainlink fence in code which can't be leapt over. If there's a better flag to use here LMK